src/ngraph/serializer.cpp

```diff
@@ -458,6 +458,37 @@ std::string ngraph::serialize(std::shared_ptr<ngraph::Function> func, size_t ind
     return ::serialize(func, indent, false);
 }
 
+template <typename T>
+std::string ngraph::serialize_attrs(const std::vector<std::shared_ptr<T>>& nodes)
```

Comment:
It's best if the serialize and deserialize have the same kinds of arguments.
Call it `serialize_types` and have it take a `const std::vector<std::pair<PartialShape, element::Type>>&` argument.

Reply:
agree..will make the change.